src/DurableTask.Netherite/Scaling/LoadPublishWorker.cs

```diff
@@ -16,19 +16,26 @@ class LoadPublishWorker : BatchWorker<(uint, PartitionLoadInfo)>
 
         // we are pushing the aggregated load information on a somewhat slower interval
         public static TimeSpan AggregatePublishInterval = TimeSpan.FromSeconds(2);
-        readonly CancellationTokenSource cancelWait = new CancellationTokenSource();
+        TaskCompletionSource<object> cancelWait = new TaskCompletionSource<object>();
 
-        public LoadPublishWorker(ILoadPublisherService service, CancellationToken token, OrchestrationServiceTraceHelper traceHelper) : base(nameof(LoadPublishWorker), false, int.MaxValue, token, null)
+        public LoadPublishWorker(ILoadPublisherService service, OrchestrationServiceTraceHelper traceHelper) : base(nameof(LoadPublishWorker), false, int.MaxValue, CancellationToken.None, null)
         {
             this.service = service;
             this.traceHelper = traceHelper;
-            this.cancelWait = new CancellationTokenSource();
         }
 
         public Task FlushAsync()
         {
-            this.cancelWait.Cancel(); // so that we don't have to wait the whole delay
-            return this.WaitForCompletionAsync();
+            var task = this.WaitForCompletionAsync();
+            this.CancelCurrentWait();
```

Comment:
nit - can we flip the order of these two? It makes more sense to me that first we'd cancel the task, and then wait for completion

Reply:
Flipping the order would create a potential race: if calling it first, the `CancelCurrentWait` could cancel the current wait, the worker sees that there is no work, and starts waiting again (the worker is in an infinite loop that waits between iterations, and the cancel does not cancel the worker, just the wait). Now we call `WaitForCompletionAsync` and it has to wait for the whole duration because the cancel already canceled the previous wait.

In the current order, we are guaranteed that when we call `WaitForCompletionAsync`, the task has either already completed, or is queued for execution behind the wait.

(I think this was one of the bugs fixed in this PR).

Reply:
Gotcha, that makes sense.
This seems subtle. Can you please add a comment here explaining the risk of this race? Otherwise, I worry we'll forget and may swap this in the future doing a refactoring.